### DISCUSSION TITLE
script to get all icons of a branch

### DISCUSSION THREAD

**Discussion by EdoFro** [freeplane/freeplane#409] [Show and tell]
Today I wanted to know which icons I used to use in an old mindmap that I haven't use for almost three years.
My first thought was that it was a difficult task, but I opened the script editor and with two lines of code I had the solution I needed.

Here it is. Maybe it is helpful for others too.

```groovy
def usedIcons = node.findAll()*.icons*.icons.flatten().unique().sort()
node.createChild("<-- present icons in '${node.text}'").icons.addAll(usedIcons)

```

**Comment by abc16361** [comment]
Thank you! I would keep it in my files.
